src/framework/jobState/FileSystemJobState/flushDataToDisk.ts

```diff
@@ -0,0 +1,91 @@
+import pMap from 'p-map';
+import { v4 as uuid } from 'uuid';
+import groupBy from 'lodash/groupBy';
+
+import { writeJsonToPath, symlink } from '../../../cacheDirectory';
+
+import { Entity, Relationship } from '../../types';
+
+export type CollectionType = 'entities' | 'relationships';
+
+interface FlushDataToDiskInput {
+  cacheDirectory?: string;
+  step: string;
+  collectionType: CollectionType;
+  data: Entity[] | Relationship[];
+}
+
+/**
+ * Utility for flushing collected data to disk and
+ * creating a symlink in the 'index' directory
+ * based on the entity or relationship '_type'.
+ */
+export async function flushDataToDisk({
+  cacheDirectory,
+  step,
+  collectionType,
+  data,
+}: FlushDataToDiskInput) {
```

Comment:
If the type for the object parameter is so closely scoped that its name is `nameOfFunc + "Input"`, why don't you simply include it in the function declaration?